app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -6,4 +6,21 @@ def customize_destination
 
     signal :post_create_destination
   end
+
+  def clone_failure_cleanup
+    delete_instance(phase_context[:clone_options])
+  end
+
+  def delete_instance(instance_attrs)
+    source.with_provider_connection do |azure|
+      nis = ::Azure::Armrest::Network::NetworkInterfaceService.new(azure)
+      ips = ::Azure::Armrest::Network::IpAddressService.new(azure)
+      ni = nis.get(instance_attrs[:name], resource_group.name)
+      ip = ips.get("#{instance_attrs[:name]}-publicIp", resource_group.name)
+      nis.delete(instance_attrs[:name], resource_group.name) if ni
+      ips.delete("#{instance_attrs[:name]}-publicIp", resource_group.name) if ip
+    end
+  rescue
```

Comment:
Was there a specific exception that Azure raised when you weren't able to delete these ?

Reply:
Yeah, this was also a Azure::Armrest::BadRequestException, so I can make it
`rescue Azure::Armrest::BadRequestException`